modules/deployment-scripts/create-agw-kv-certificate/main.bicep

```diff
@@ -0,0 +1,229 @@
+@description('The name of the Azure Key Vault')
+param akvName string
+
+@description('The name of the Azure Application Gateway')
+param agwName string = ''
+
+@description('The location to deploy the resources to')
+param location string = resourceGroup().location
+
+@description('How the deployment script should be forced to execute')
+param forceUpdateTag  string = utcNow()
+
+@description('Version of the Azure CLI to use')
+param azCliVersion string = '2.30.0'
+
+@description('Deployment Script timeout')
+param timeout string = 'PT30M'
+
+@description('The retention period for the deployment script')
+param retention string = 'P1D'
+
+@description('An array of Azure Key Vault RoleIds that are required for the DeploymentScript resource')
+param rbacRolesNeededOnKV array = [
+  'a4417e6f-fecd-4de8-b567-7b0420556985' //KeyVault Certificate Officer
+]
+
+@description('An array of Azure Application Gateway RoleIds that are required for the DeploymentScript resource')
+param rbacRolesNeededOnAppGw array = [
+  'b24988ac-6180-42a0-ab88-20f7382dd24c' //Contributor
+]
+
+@description('Does the Managed Identity already exists, or should be created')
+param useExistingManagedIdentity bool = false
+
+@description('Name of the Managed Identity resource')
+param managedIdentityName string = 'id-KeyVaultCertificateCreator'
+
+@description('For an existing Managed Identity, the Subscription Id it is located in')
+param existingManagedIdentitySubId string = subscription().subscriptionId
+
+@description('For an existing Managed Identity, the Resource Group it is located in')
+param existingManagedIdentityResourceGroupName string = resourceGroup().name
+
+@description('An array of Certificate names to create')
+param certNames array
+
+@allowed([
+  'none'
+  'root-cert'
+  'ssl-cert'
+])
+@description('Configured certificate in Application Gateway as Frontend (ssl-cert) or Backend (root-cert)')
+param agwCertType string = 'none'
+
+@description('A delay before the script import operation starts. Primarily to allow Azure AAD Role Assignments to propagate')
+param initialScriptDelay string = '30s'
+
+@description('A delay before the Application Gateway imports the Certificate from KeyVault. Primarily to allow the certificate creation operation to complete')
+param agwCertImportDelay string = '120s'
+
+@allowed([
+  'OnSuccess'
+  'OnExpiration'
+  'Always'
+])
+@description('When the script resource is cleaned up')
+param cleanupPreference string = 'OnSuccess'
+
+resource akv 'Microsoft.KeyVault/vaults@2021-11-01-preview' existing = {
+  name: akvName
+}
+
+resource agw 'Microsoft.Network/applicationGateways@2021-05-01' existing = if (!empty(agwName) && agwCertType != 'none') {
```

Comment:
this allowed the module to function to just create certificates.
since this module is going to be downscoped to just KV, i'll resolve this discussion